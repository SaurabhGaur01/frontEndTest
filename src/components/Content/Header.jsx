import React from 'react';

const Header = () => (
    <header className="appHeader">
        <div className="d-flex justify-content-between">
            <h3 className="title text-white pl-3 pt-3">Find Your Car</h3>
            <div className="pt-3 pr-3"><button type="button" className="btn btn-light">Manage Bookings</button></div>
        </div>
        <div className="text-center pb-4">
            <h3 className="title text-white pl-3 pt-3">Car Hire - Search, Compare & Save</h3>
        </div>
    </header>
);

export default Header;