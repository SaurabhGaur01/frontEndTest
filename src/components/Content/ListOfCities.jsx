import React from 'react';
import PropTypes from 'prop-types';
import { transformStrings, isAirportOrCity } from '../../utils/index';

const DOCS_KEY_FIELD = 'docs';
const CITY_BOOKING = 'city';
const AIRPORT_BOOKING = 'airport';

const ListOfCities = ({ listOfCities, keyFields, getSelectedCity }) => {
    if(keyFields === DOCS_KEY_FIELD) {
        return listOfCities[keyFields].map(item => {
            const { name, index, bookingId, city, region, country } = item;
            return (
                <li className="listingItem ml-0" key={index} onClick={() =>getSelectedCity(item)}>
                    <div className="d-flex pb-3">
                        {
                            isAirportOrCity(bookingId) === CITY_BOOKING && (
                                <span className="badge bg-primary mr-3 mt-2 iconDisplay">City</span>
                            )
                        }
                        {
                            isAirportOrCity(bookingId) === AIRPORT_BOOKING && (
                                <span className="badge bg-warning mr-3 mt-2 iconDisplay">Airport</span>
                            )
                        }
                        <div className="details">
                            <p className="m-0">{name}</p>
                            <p className="m-0">{transformStrings(city, region, country)}</p>
                        </div>
                    </div>
                </li> 
            )
        });
    }

    return null;
}

ListOfCities.propTypes = {
    keyFields: PropTypes.string.isRequired,
    getSelectedCity: PropTypes.func.isRequired,
    listOfCities: PropTypes.shape({
        docs: PropTypes.arrayOf(PropTypes.shape({
            name: PropTypes.string, 
            index: PropTypes.number, 
            bookingId: PropTypes.string,
            city: PropTypes.string,
            region: PropTypes.string,
            country: PropTypes.string, 
        })),
    }).isRequired,
}

export default ListOfCities;