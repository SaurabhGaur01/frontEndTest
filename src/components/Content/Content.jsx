import React from 'react';
import PropTypes from 'prop-types';
import ListOfCities from './ListOfCities';

const Content = ({ 
    onCityChange, 
    selectedCity, 
    listOfCities, 
    getSelectedCity 
}) => (
    <div className="section">
        <div className="search-bar">
            <div className="search-bar__background p-1">
                <div className="searchbox">
                    <input type="search" id="pickup-location" className="form-control textBox" placeholder="Pickup Location"
                        aria-label="Search" onChange={onCityChange} value={selectedCity} />
                    {
                        Object.entries(listOfCities).length > 0 && (
                            <div className="card searchbox__suggestions">
                                <ul className="searchbox__listing pt-2">
                                    {
                                        Object.keys(listOfCities).map((item, index) => {
                                            return (
                                                <ListOfCities 
                                                    key={index} 
                                                    listOfCities={listOfCities} 
                                                    keyFields={item} 
                                                    getSelectedCity={getSelectedCity}
                                                />
                                            )
                                        })
                                    }
                                </ul>
                            </div>   
                        )
                    } 
                </div>
            </div>
        </div>
        <div className="backSection" />
    </div>
);

Content.propTypes = {
    onCityChange: PropTypes.func.isRequired,
    selectedCity: PropTypes.string.isRequired,
    getSelectedCity: PropTypes.func.isRequired,
    listOfCities: PropTypes.shape({}).isRequired,
}

export default Content;