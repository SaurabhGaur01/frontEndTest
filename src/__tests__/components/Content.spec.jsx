import React from 'react';
import { shallow } from 'enzyme';
import Content from '../../components/Content/Content';

describe('Content', () => {
    let props;

    beforeEach(() =>{
        props = {
            onCityChange: jest.fn(),
            getSelectedCity: jest.fn(),
            listOfCities: {
                mockKey: 'mockValue',
            },
            selectedCity: 'mockCity',

        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<Content  {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should render correctly when listOfCities is empty', () => {
        props.listOfCities = {};
        const renderedModule = shallow(<Content  {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should call onCityChange when input changes', () => {
        const renderedModule = shallow(<Content  {...props} />);
        renderedModule.find('#pickup-location').at(0).simulate('change', {target: { value: 'mockValue'}});
        expect(props.onCityChange).toHaveBeenCalledTimes(1);
    });
})