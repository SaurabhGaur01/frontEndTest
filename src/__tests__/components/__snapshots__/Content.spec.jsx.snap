// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render correctly 1`] = `
<div
  className="section"
>
  <div
    className="search-bar"
  >
    <div
      className="search-bar__background p-1"
    >
      <div
        className="searchbox"
      >
        <input
          aria-label="Search"
          className="form-control textBox"
          id="pickup-location"
          onChange={[MockFunction]}
          placeholder="Pickup Location"
          type="search"
          value="mockCity"
        />
        <div
          className="card searchbox__suggestions"
        >
          <ul
            className="searchbox__listing pt-2"
          >
            <ListOfCities
              getSelectedCity={[MockFunction]}
              key="0"
              keyFields="mockKey"
              listOfCities={
                Object {
                  "mockKey": "mockValue",
                }
              }
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="backSection"
  />
</div>
`;

exports[`Content should render correctly when listOfCities is empty 1`] = `
<div
  className="section"
>
  <div
    className="search-bar"
  >
    <div
      className="search-bar__background p-1"
    >
      <div
        className="searchbox"
      >
        <input
          aria-label="Search"
          className="form-control textBox"
          id="pickup-location"
          onChange={[MockFunction]}
          placeholder="Pickup Location"
          type="search"
          value="mockCity"
        />
      </div>
    </div>
  </div>
  <div
    className="backSection"
  />
</div>
`;
